extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java

```diff
@@ -341,6 +313,106 @@ void openTracingIntegration(Capabilities capabilities,
+            }
+            systemProperties.produce(new SystemPropertyBuildItem(ConfigKey.ENABLE_METRICS, TRUE));
+        } else {
+            systemProperties.produce(new SystemPropertyBuildItem(ConfigKey.ENABLE_METRICS, FALSE));
+        }
+    }
+
+    @BuildStep
+    void activateTracing(Capabilities capabilities,
+            BuildProducer<SystemPropertyBuildItem> systemProperties) {
+
+        boolean activate = shouldActivateService(capabilities, quarkusConfig.tracingEnabled, Capability.OPENTRACING,
+                "quarkus.smallrye-graphql.tracing.enabled");
+        if (activate) {
+            systemProperties.produce(new SystemPropertyBuildItem(ConfigKey.ENABLE_TRACING, TRUE));
+        } else {
+            systemProperties.produce(new SystemPropertyBuildItem(ConfigKey.ENABLE_TRACING, FALSE));
+        }
+    }
+
+    @BuildStep
+    void activateValidation(Capabilities capabilities,
+            BuildProducer<SystemPropertyBuildItem> systemProperties) {
+
+        boolean activate = shouldActivateService(capabilities, quarkusConfig.validationEnabled, Capability.HIBERNATE_VALIDATOR,
+                "quarkus.smallrye-graphql.validation.enabled");
+        if (activate) {
+            systemProperties.produce(new SystemPropertyBuildItem(ConfigKey.ENABLE_VALIDATION, TRUE));
+        } else {
+            systemProperties.produce(new SystemPropertyBuildItem(ConfigKey.ENABLE_VALIDATION, FALSE));
+        }
+    }
+
+    @BuildStep
+    void activateEventing(BuildProducer<SystemPropertyBuildItem> systemProperties) {
+        if (quarkusConfig.eventsEnabled) {
+            systemProperties.produce(new SystemPropertyBuildItem(ConfigKey.ENABLE_EVENTS, TRUE));
+        } else {
+            systemProperties.produce(new SystemPropertyBuildItem(ConfigKey.ENABLE_EVENTS, FALSE));
+        }
+    }
+
+    private boolean shouldActivateService(Capabilities capabilities,
+            Optional<Boolean> serviceEnabled,
+            Capability linkedCapability,
+            String configKey) {
+
+        return shouldActivateService(capabilities, serviceEnabled, capabilities.isPresent(linkedCapability),
+                linkedCapability.getName(), configKey);
+    }
+
+    private boolean shouldActivateService(Capabilities capabilities,
+            Optional<Boolean> serviceEnabled,
+            boolean linkedCapabilityIsPresent,
+            String linkedCapabilityName,
+            String configKey) {
+
+        if (serviceEnabled.isPresent()) {
+            // The user explisitly asked from something
+            boolean isEnabled = serviceEnabled.get();
+            if (isEnabled) {
+                if (linkedCapabilityIsPresent) {
+                    // enable
+                    return true;
+                } else {
+                    // Warn and disable
+                    LOG.warnf(SERVICE_NOT_AVAILABLE_WARNING, configKey, linkedCapabilityName, linkedCapabilityName);
```

Comment:
I think we should have a proper parameter to passe the extension name and not the capability.